modules/telestion-api/src/main/java/de/wuespace/telestion/api/utils/AbstractUtils.java

```diff
@@ -0,0 +1,25 @@
+package de.wuespace.telestion.api.utils;
+
+/**
+ * When class extends AbstractUtils it indicates that the extending class is a utility class which can not be
+ * instantiated. There should be no callable constructors which is why the default constructor is also blocked.
+ *
+ * @author Cedric Boes (@cb0s), Ludwig Richter (@fussel178)
+ */
+public abstract class AbstractUtils {
+	/**
+	 * There shall be no objects of a Utility class. This is why this constructor throws an
+	 * {@link UnsupportedOperationException}.
+	 *
+	 * @throws UnsupportedOperationException Will always be thrown because there shall be no objects of a utility class
+	 */
+	// Utility classes do not have objects
+	protected AbstractUtils() throws UnsupportedOperationException {
+		var className = this.getClass().getName();
+		var errorMsg = String.format("""
+				There shall be no objects of a utility class!
+				%s extending AbstractUtils is automatically a utility class. It must not be instantiated.""",
+				className);
+		throw new UnsupportedOperationException(errorMsg);
+	}
+}
```

Comment:
No. Just no. If you want to do this, then make the constructor private and everything is done. But you don't want that. Everyone should be able to write a utility class. I would never guess that I could use this class. In the end, I will look at the subclass and ask myself: Why the heck does a utility class have a base class and where is the private constructor?

Reply:
We proposed this, because we could not think of a solution of how to implement this with a private constructor. With a private constructor, obviously no inheriting classes can be existent, as they would have to call the constructor somehow... If we do it protected, the API-user still can't call the constructor but can anticipate, that this cannot be instantiated. We wanted to have a base class for the API-utils, which can be used to identify those and to ideally reduce boiler plate code, as no private default constructor is needed any more in the implementation of the different utils.

Do you have a better idea of how to solve this problem? :)